app/models/host/managed.rb

```diff
@@ -778,6 +779,18 @@ def setup_clone
 
   private
 
+  # validate uniqueness can't prevent saving two interfaces that has same DNS name
+  # because the validation happens before transaction is committed, so data are not in DB
+  # yet, this is the reason why we "reimplement" uniqueness validation
+  def validate_dns_name_uniqueness
+    dups = self.interfaces.group_by { |i| [ i.name, i.domain_id ] }.detect { |_, nics| nics.count > 1 }
```

Comment:
Good catch, also I wanted to ask if we should run this validation only on managed interfaces (including managed host and unattended mode) only it might be OK to have same FQDN on two unmanaged interfaces. OTOH all of these flags can be changed in time, some of them making host invalid.